llvm/test/CodeGen/SPIRV/ShaderBufferImage.ll

```diff
@@ -0,0 +1,20 @@
+; RUN: llc -verify-machineinstrs -O0 -mtriple=spirv-vulkan-library %s -o - | FileCheck %s
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv-vulkan-library %s -o - -filetype=obj | spirv-val %}
+
+; CHECK-NOT: OpCapability ImageBasic
+; CHECK-NOT: OpCapability ImageReadWrite
+; CHECK: OpCapability ImageBuffer
+; CHECK-NOT: OpCapability ImageBasic
+; CHECK-NOT: OpCapability ImageReadWrite
+
+; CHECK: [[Float:%[0-9]+]] = OpTypeFloat 32
```

Comment:
We need `CHECK-DAG` applied to `OpType...` everywhere, in this and the next test case, because SPIRV spec doesn't guarantee the order of those type instructions and implementation details can change anytime.